case-lib/user.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# function func_check_version uses dpkg to get the package version
+# and compares it with the specified version. If the package version
+# is less than the specified version, it means the version dependency
+# is not met and exit with die.
+# Args: $1: package name
+#       $2: specifiedversion
+func_check_version()
+{
+    ver=$(dpkg -l "$1" |grep "$1" | awk '{print $3}')
+    [[ -z "$ver" ]] && die "failed to get $1 version"
+    dlogi "find $1 version  $ver"
+    if [ "$(printf '%s\n' "$2" "$ver" | sort -V | head -n1)" = "$ver" ]; then
+	die "$1 version $ver doesn't meet the requirement >= $2"
+    fi
+}
+
+# function func_version_ge uses dpkg to get the package version
+# and compares it with the specified version. If the package version
+# is less than the specified version, it means the version dependency
+# is not met and return 1.
+# Args: $1: package name
+#       $2: specifiedversion
+func_version_ge()
+{
+    ver=$(dpkg -l "$1" |grep "$1" | awk '{print $3}')
+    [[ -z "$ver" ]] && die "failed to get $1 version"
+    dlogi "find $1 version  $ver"
+    if [ "$(printf '%s\n' "$2" "$ver" | sort -V | head -n1)" = "$ver" ]; then
+	return 1
```

Comment:
You don't need `return`, the return value of a function is the return value of the last command. Just run `[ ]` (or `test` ) last.

Reply:
ACK